### PR TITLE
Feat/web login

### PR DESCRIPTION
**Pull request by Leon-pix-coder**
Add Web login

### CODE REVIEW

File: src/components/Community/components/Archived/index.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  }, [type]);
+
+  useEffect(() => {
+    if (data) {
+      setSections((prev) =>
+        pageIndex === 0 ? data.data : [...prev, ...data.data]
+      );
+      setHasMore(data.data?.length >= PAGE_SIZE);
+    }
+  }, [data, pageIndex]);
```
Comment (evan-spidy) [inline src/components/Community/components/Archived/index.tsx]: use better name, do not use data.data

File: src/components/Community/index.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+interface ICommunityProps {
+  scrollTop: number;
+}
+
+const Community = ({ scrollTop }: ICommunityProps) => {
+  const [currentTab, setCurrentTab] = useState(1);
+
+  return (
+    <div>
```
Comment (evan-spidy) [inline src/components/Community/index.tsx]: what is the reason for additional div?

File: src/components/Community/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      <ToggleSlider
+        current={currentTab}
+        items={["Archived", "Current"]}
```
Comment (evan-spidy) [inline src/components/Community/index.tsx]: make this a constants

File: src/components/ForYou/AppDetail/index.tsx [hunk 4]
```diff
@@ -76,10 +76,10 @@ const AppDetail = ({ item, updateOpenAppClick }: IAppDetailProps) => {
             alt=""
           />
           <div className="flex flex-col gap-[5px] justify-center">
-            <span className="font-outfit font-bold text-[16px] leading-[16px]">
+            <span className="font-outfit font-bold text-[16px] leading-[16px] text-white">
               {item.title}
             </span>
-            <span className="text-[11px] leading-[13px]">
+            <span className="text-[11px] leading-[13px] text-white">
```
Comment (evan-spidy) [inline src/components/ForYou/AppDetail/index.tsx]: if both need text-white, put it at the div to set text-white

File: src/components/SimpleDatePicker/index.tsx [hunk 5]
```diff
@@ -9,16 +9,14 @@ interface ISimpleDatePickerProps {
   value?: string;
   defaultVulue?: string;
   className?: string;
-  onChange?(t: string): void;
   onChange?: (value: string) => void;
 }
 
 const SimpleDatePicker = (props: ISimpleDatePickerProps) => {
   const { value, defaultVulue, className, onChange, ...dayPickerProps } = props;
+  const baseValue = value && dayjs(value || "").isValid() ? value : defaultVulue && dayjs(defaultVulue || "").isValid() ? defaultVulue : dayjs().format("")
```
Comment (evan-spidy) [inline src/components/SimpleDatePicker/index.tsx]: check your spelling defaultVulue

File: src/components/SimpleTimePicker/index.tsx [hunk 6]
```diff
@@ -11,19 +11,18 @@ import clsx from "clsx";
 
 interface ISimpleTimePickerProps {
   className?: string;
-  value?: string;
-  defaultVulue?: string;
-  onChange?(t: string): void;
+  value?: string | number;
+  onChange?(t: number): void;
```
Comment (evan-spidy) [inline src/components/SimpleTimePicker/index.tsx]: give me a better name what is t ?

File: src/components/TMAs/components/Accumulative/index.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      search,
+    }).toString()}`
+  );
+
+  useEffect(() => {
+    if (data) {
+      setVoteList((prev) =>
+        pageIndex === 0 ? data.data : [...prev, ...data.data]
+      );
+      setHasMore(data.data?.length >= PAGE_SIZE);
```
Comment (evan-spidy) [inline src/components/TMAs/components/Accumulative/index.tsx]: do not use data.data

File: src/components/TMAs/components/Accumulative/index.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    }
+  }, [hasMore, isLoading, scrollTop]);
+
+  useEffect(() => {
+    setSearch(keyward || '');
+    setPageIndex(0);
+  }, [keyward]);
+
+  useEffect(() => {
+    setCategory(cate || '');
```
Comment (evan-spidy) [inline src/components/TMAs/components/Accumulative/index.tsx]: what is cate?

File: src/components/ToggleSlider/index.tsx [hunk 9]
```diff
@@ -17,30 +30,54 @@ const ToggleSlider = ({ items = [], onChange }: IToggleSlider) => {
...
+  // Ensure layout is ready before running animations
+  useEffect(() => {
+    // Wait until the next frame to allow the DOM to fully render
+    const timeout = setTimeout(() => {
+      setReady(true);
+    }, 0);
+
+    return () => clearTimeout(timeout);
+  }, [items.length]);
+
```
Comment (evan-spidy) [inline src/components/ToggleSlider/index.tsx]: i will approve now, i think can be written better.

File: src/components/Vote/index.tsx [hunk 10]
```diff
@@ -1,107 +1,69 @@
...
-import { voteSectionData, VoteListItems } from "@/__mocks__/VoteApp";
+import TMAs from "../TMAs";
+import { useCallback, useEffect, useRef, useState } from "react";
+import Community from "../Community";
 
 const Vote = () => {
   const {
     user: { userPoints },
   } = useUserContext();
+  const [currnetTab, setCurrentTab] = useState("TMAs");
```
Comment (evan-spidy) [inline src/components/Vote/index.tsx]: use constants

File: src/components/Vote/index.tsx [hunk 11]
```diff
@@ -1,107 +1,69 @@
...
+      className="h-screen overflow-scroll pt-telegramHeader bg-black"
+      ref={scrollViewRef}
+    >
       <div className="votigram-grid">
         <div className="col-7 h-7">
-          <ToggleSlider items={["TMAs", "Community"]} />
+          <ToggleSlider
+            items={["TMAs", "Community"]}
+            onChange={(index) =>
+              setCurrentTab(index === 0 ? "TMAs" : "Community")
```
Comment (evan-spidy) [inline src/components/Vote/index.tsx]: same here, use constants

File: src/components/VoteItem/index.tsx [hunk 12]
```diff
@@ -102,38 +235,89 @@ const VoteItem = ({
...
           type="button"
+          ref={buttonRef}
           className="bg-white/[.25] w-[40px] h-[40px] flex justify-center items-center p-[8px] rounded-[20px] shrink-0 z-[10]"
           onClick={onVoteClick}
         >
           <i
             className={clsx(
               "votigram-icon-navbar-vote text-[24px]",
-              data?.isVoted ? "text-lime-green" : "text-lime-primary"
+              isFirst ? "text-lime-green" : "text-lime-primary"
```
Comment (evan-spidy) [inline src/components/VoteItem/index.tsx]: use CSS to set as first-child

File: src/config/mainet.ts
Comment (evan-spidy) [inline src/config/mainet.ts:1]: the file name should be mainnet

File: src/config/mainet.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+export const connectServer = 'https://auth-portkey.portkey.finance';
+export const graphqlServer = 'https://indexer-api.aefinder.io/api/app/graphql/portkey';
+export const portkeyServer = 'https://aa-portkey.portkey.finance';
+export const connectUrl = 'https://auth-aa-portkey.portkey.finance';
+export const sideChainCAContractAddress = '2UthYi7AHRdfrqc1YCfeQnjdChDLaas65bW4WxESMGMojFiXj9';
+export const propalAddress = '2tCM3oV6dTCmwFxSiFGPEVhGngdMwBV741wi156vj8kmqfp6da';
+export const voteAddress = '2A8h4hLynLt86RxqvpNY43x6Js8CYhgyuAzj7sDGQ2ecP77Zgp';
+export const curChain = 'tDVW';
+
+export const host = 'https://test.tmrwdao.com';
```
Comment (evan-spidy) [inline src/config/mainet.ts]: mainnet still use test?

File: src/contract/proposalCreateContract.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+import { formatErrorMsg } from './util';
+import { getTxResult } from '@/utils/getTxResult';
+
+export const proposalCreateContractRequest = async <T>(
+  methodName: string,
+  params: T,
+  options?: IContractOptions,
+): Promise<IContractResult> => {
+  const contractAddress = propalAddress;
+  console.log('proposalCreateContractRequest', contractAddress);
```
Comment (evan-spidy) [inline src/contract/proposalCreateContract.ts]: remove console.log

File: src/contract/proposalCreateContract.ts [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+  try {
+    if (options?.type === ContractMethodType.VIEW) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const res: { data: any } = await webLoginInstance.callViewMethod(chainId, {
```
Comment (evan-spidy) [inline src/contract/proposalCreateContract.ts]: do not declare any

File: src/contract/proposalCreateContract.ts [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+        contractAddress,
+        methodName,
+        args: params,
+      });
+      const result = res.data as unknown as IContractError;
+      console.log('webLoginInstance', result);
```
Comment (evan-spidy) [inline src/contract/proposalCreateContract.ts]: remove console.log

File: src/contract/proposalCreateContract.ts [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+      console.log('proposalCreateContractRequest', res);
+      const result = res as unknown as IContractError;
+      console.log('webLoginInstance', result);
+      if (result?.error || result?.code || result?.Error) {
+        throw formatErrorMsg(result);
+      }
+
+      const { transactionId, TransactionId } = result.result || result;
+      const resTransactionId = TransactionId || transactionId;
+      await sleep(1000);
```
Comment (evan-spidy) [inline src/contract/proposalCreateContract.ts:42]: do not use sleep, use setTimeout

File: src/contract/util.ts [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    return resMessage.replace('AElf.Sdk.CSharp.AssertionException: ', '');
+  } else {
+    return DEFAULT_ERROR;
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const stringifyMsg = (message: any) => {
```
Comment (evan-spidy) [inline src/contract/util.ts]: do not declare any

File: src/contract/webLogin.ts
Comment (evan-spidy) [inline src/contract/webLogin.ts:1]: i will pause here. Will continue to review tmr

File: src/contract/util.ts (hunk 18 above)
Comment (Leon-pix-coder) [reply src/contract/util.ts]: Updated

File: src/contract/proposalCreateContract.ts (hunk 17 above)
Comment (Leon-pix-coder) [reply src/contract/proposalCreateContract.ts:42]: It's setTimeout

File: src/contract/proposalCreateContract.ts (hunk 16 above)
Comment (Leon-pix-coder) [reply src/contract/proposalCreateContract.ts]: Removed

File: src/contract/proposalCreateContract.ts (hunk 15 above)
Comment (Leon-pix-coder) [reply src/contract/proposalCreateContract.ts]: Updated

File: src/config/mainet.ts (hunk 13 above)
Comment (Leon-pix-coder) [reply src/config/mainet.ts]: Updated

File: src/config/mainet.ts
Comment (Leon-pix-coder) [reply src/config/mainet.ts:1]: Updated

File: src/components/VoteItem/index.tsx (hunk 12 above)
Comment (Leon-pix-coder) [reply src/components/VoteItem/index.tsx]: Updated

File: src/components/Community/components/Archived/index.tsx (hunk 1 above)
Comment (Leon-pix-coder) [reply src/components/Community/components/Archived/index.tsx]: Updated

File: src/components/Community/index.tsx (hunk 3 above)
Comment (Leon-pix-coder) [reply src/components/Community/index.tsx]: Updated

File: src/components/ForYou/AppDetail/index.tsx (hunk 4 above)
Comment (Leon-pix-coder) [reply src/components/ForYou/AppDetail/index.tsx]: Updated

File: src/components/SimpleDatePicker/index.tsx (hunk 5 above)
Comment (Leon-pix-coder) [reply src/components/SimpleDatePicker/index.tsx]: Updated

File: src/components/SimpleTimePicker/index.tsx (hunk 6 above)
Comment (Leon-pix-coder) [reply src/components/SimpleTimePicker/index.tsx]: Updated

File: src/components/TMAs/components/Accumulative/index.tsx (hunk 7 above)
Comment (Leon-pix-coder) [reply src/components/TMAs/components/Accumulative/index.tsx]: Updated

File: src/components/TMAs/components/Accumulative/index.tsx (hunk 8 above)
Comment (Leon-pix-coder) [reply src/components/TMAs/components/Accumulative/index.tsx]: Updated

File: src/components/Vote/index.tsx (hunk 11 above)
Comment (Leon-pix-coder) [reply src/components/Vote/index.tsx]: Updated

File: src/components/Community/index.tsx (hunk 2 above)
Comment (Leon-pix-coder) [reply src/components/Community/index.tsx]: Removed

File: src/contract/proposalCreateContract.ts (hunk 14 above)
Comment (Leon-pix-coder) [reply src/contract/proposalCreateContract.ts]: Removed

File: src/components/Vote/index.tsx (hunk 10 above)
Comment (Leon-pix-coder) [reply src/components/Vote/index.tsx]: Updated

File: src/components/ToggleSlider/index.tsx (hunk 9 above)
Comment (Leon-pix-coder) [reply src/components/ToggleSlider/index.tsx]: Removed

File: src/contract/webLogin.ts [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+  chainId: string;
+}
+
+export default class WebLoginInstance {
+  public contract: unknown;
+  public address: string | undefined;
+  public chainId: string | undefined;
+
+  private static instance: WebLoginInstance | null = null;
+  private context: any | null = null;
```
Comment (evan-spidy) [inline src/contract/webLogin.ts]: do not add any

File: src/pageComponents/CreatePoll/index.tsx [hunk 20]
```diff
@@ -9,69 +9,337 @@ import ToggleSlider from "@/components/ToggleSlider";
...
+  proposalTitle: [
+    (value: string) => (!value ? "Please Enter Topic" : undefined),
+  ],
+  options: [
+    (ops: VoteOption[]) =>
+      ops.length === 0
+        ? "Please add an option"
+        : ops.length < 2
+        ? "Please add at least two options"
+        : undefined,
```
Comment (evan-spidy) [inline src/pageComponents/CreatePoll/index.tsx]: split them into 2 if loop

File: src/pageComponents/CreatePoll/index.tsx [hunk 21]
```diff
@@ -9,69 +9,337 @@ import ToggleSlider from "@/components/ToggleSlider";
...
+              chainId,
+              daoId: communityDaoId ?? "",
+            }).toString()}`
+          ),
+        ]);
+      const appAlias = saveRes?.data ?? [];
+      if (!appAlias.length) {
+        throw new Error("Failed to create proposal, save options failed");
+      }
+      const formmatDescriptionStr = formmatDescription(
```
Comment (evan-spidy) [inline src/pageComponents/CreatePoll/index.tsx]: wrong spelling here

File: src/pageComponents/CreatePoll/index.tsx (hunk 20 above)
Comment (Leon-pix-coder) [reply src/pageComponents/CreatePoll/index.tsx]: split them with tow rule

File: src/pageComponents/CreatePoll/index.tsx (hunk 21 above)
Comment (Leon-pix-coder) [reply src/pageComponents/CreatePoll/index.tsx]: updated

File: src/contract/webLogin.ts (hunk 19 above)
Comment (Leon-pix-coder) [reply src/contract/webLogin.ts]: Updated

**Pull request merged**